src/routes_admin.py

```diff
@@ -0,0 +1,237 @@
+            except json.decoder.JSONDecodeError:
+                return util.error("Failed to decode config JSON")
+
+            visibility = request.form["visibility"] == "visible"
+
+            db.add_assignment(cid, aid, max_runs, quota, start, end, visibility)
+            return "", 204
+
+        @blueprint.route("/staff/course/<cid>/<aid>/edit/", methods=["POST"])
+        @auth.require_auth
+        @auth.require_admin_status
+        def staff_edit_assignment(netid, cid, aid):
+            course = db.get_course(cid)
+            assignment = db.get_assignment(cid, aid)
+            if course is None or assignment is None:
+                return abort(404)
+
+            missing = util.check_missing_fields(request.form, *["max_runs", "quota", "start", "end", "visibility"])
+            if missing:
+                return util.error("Missing fields (%s)." % (", ".join(missing)))
+
+            try:
+                max_runs = int(request.form["max_runs"])
+                if max_runs < 1:
+                    return util.error("Max Runs must be a positive integer.")
+            except ValueError:
+                return util.error("Max Runs must be a positive integer.")
+
+            quota = request.form["quota"]
+            if not db.Quota.is_valid(quota):
+                return util.error("Quota Type is invalid.")
+
+            start = util.parse_form_datetime(request.form["start"]).timestamp()
+            end = util.parse_form_datetime(request.form["end"]).timestamp()
+            if start is None or end is None:
+                return util.error("Missing or invalid Start or End.")
+            if start >= end:
+                return util.error("Start must be before End.")
+
+            try:
+                config_str = request.form.get("config")
+
+                if config_str is not None:  # skip update otherwise
+                    config = json.loads(request.form["config"])
+                    msg = bw_api.set_assignment_config(cid, aid, config)
+
+                    if msg:
+                        return util.error("Failed to add assignment to Broadway: {}".format(msg))
+            except json.decoder.JSONDecodeError:
+                return util.error("Failed to decode config JSON")
+
+            visibility = request.form["visibility"] == "visible"
+
+            if not db.update_assignment(cid, aid, max_runs, quota, start, end, visibility):
+                return util.error("Save failed or no changes were made.")
+            return "", 204
+
+        @blueprint.route("/staff/course/<cid>/<aid>/extensions/", methods=["GET"])
+        @auth.require_auth
+        @auth.require_admin_status
+        def staff_get_extensions(netid, cid, aid):
+            extensions = list(db.get_extensions(cid, aid))
+            for ext in extensions:
+                ext["_id"] = str(ext["_id"])
+            return jsonify(extensions), 200
```

Comment:
I'm sure Flask has constants for all of the HTTP return codes - can we use those everywhere, instead?

Reply:
For sure

Reply:
Seems like we need an extra dependency (flask_api) for those constants. Should I install it?

Reply:
Sure, unless it's humongous, in which case you can define the constants yourself.

Reply:
Turns out the python default http module has all the status codes.